src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs

```diff
@@ -100,6 +104,16 @@ private IEnumerable<EndpointConfig> ReadEndpoints()
             return endpoints;
         }
 
+        private ClientCertificateMode? ReadClientCertificateMode()
```

Comment:
This only supports a global setting, not a setting per endpoint?

Reply:
I couldn't find the code which options for each endpoint is being set from config. Could you point me to it please?

Reply:
https://github.com/dotnet/aspnetcore/blob/050632502e4813e4e70344f7d469d602cb12d2a7/src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs#L98

Reply:
Uhhmm. I'm confused! `EndpointConfig` doesn't have a `ClientCertificateMode` property. Am I missing something? Is it even possible to set `ClientCertificateMode` per endpoint?

Reply:
EndpointConfig is an internal config construct, you should add ClientCertificateMode there.